packages/mdc-auto-init/index.ts

```diff
@@ -78,7 +82,8 @@ export default function mdcAutoInit(root = document, warn = CONSOLE_WARN) {
   return components;
 }
 
-mdcAutoInit.register = function(componentName, Ctor, warn = CONSOLE_WARN) {
+// tslint:disable-next-line:variable-name
+mdcAutoInit.register = function(componentName: string, Ctor: Function, warn = CONSOLE_WARN) {
```

Comment:
`Function` --> `() => any`

Reply:
Wouldn't something like `new (...args: any[]) => any` be more appropriate for a constructor? (I don't suppose there's a way to avoid using `any` here...)

Reply:
actually we should be using `unknown` according to our best practices

Reply:
```ts
interface ComponentClass {
  // tslint:disable-next-line:no-any a component can pass in anything it needs to the constructor
  new<F extends MDCFoundation>(root: Element, foundation?: F, ...args: any[]): MDCComponent<F>;
  attachTo<F extends MDCFoundation>(root: Element): MDCComponent<F>;
}

mdcAutoInit.register = function(componentName: MDCAutoInitKey, Ctor: ComponentClass, warn = CONSOLE_WARN) {
  // ...
}
```